File: sn-search-controller/src/main/webapp/js/service/governmenthotsService.js
//服务层
app.service('Governmenthots',function($http){

    //读取列表数据绑定到表单中
    this.findAll=function(){
        return $http.get('../Governmenthots/findAll.do');
    }
    //分页
    this.findPage=function(page,rows){
        return $http.get('../Governmenthots/findPage.do?page='+page+'&rows='+rows);
    }
    //查询实体
    this.findOne=function(id){
        return $http.get('../Governmenthots/findOne.do?id='+id);
    }
    //增加
    this.add=function(entity){
        return  $http.post('../Governmenthots/add.do',entity );
    }

    //删除
    this.dele=function(ids){
        return $http.get('../Governmenthots/delete.do?ids='+ids);
    }
    /*//搜索
    this.search=function(page,rows,searchEntity){
        return $http.post('../Governmenthots/search.do?page='+page+"&rows="+rows, searchEntity);
    }*/

});

File: sn-search-interface/src/main/java/com/sn/search/GovernmenthotsService.java
package com.sn.search;



import com.sn.entity.PageResult;
import com.sn.pojo.CmsGovernmenthots;

import java.util.List;

/**
 * 服务层接口
 */
public interface GovernmenthotsService {

    /**
     * 返回全部列表
     * @return
     */
    public List<CmsGovernmenthots> findAll();


    /**
     * 返回分页列表
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);


    /**
     * 增加
     */
    public void add(CmsGovernmenthots governmenthots);



    /**
     * 根据ID获取实体
     * @param id
     * @return
     */
    public CmsGovernmenthots findOne(Long id);


    /**
     * 批量删除
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 热词
     * @return
     */
    List selectName();

    /**
     * 根据用户ip获取搜索记录数据
     * @param aLong
     * @return
     */
    List GovHotIp(Object aLong);




}
